### PR TITLE
fix: handle const Type Parameters

### LINKED ISSUE #518: 🐛 Bug: `symbolHasReadonlyDeclaration` doesn't support `const` type parameters
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

For code like the following: [playground link](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEYD2A7AzgF3gMxQLngB5l0sBBAPgAoAHWKAWwLIEp4BeC+MgbgCgSmeBhBD2OFFQDeOJEgIyARrAIBGeAF9NLfgHpd8Q-AB6AfiA)

```ts
declare const fn: <const A>(param: A) => A;
const test = fn({ foo: { bar: 1 } });
//    ^? const test: { readonly foo: { readonly bar: 1; }; }
```

The object literal passed to `fn` should essentially be treated the same as if it was appended with `as const` and thus `symbolHasReadonlyDeclaration` should return `true` for the symbol of each of its deep properties.

### Actual

`symbolHasReadonlyDeclaration` returns `false` for those symbols.

### Additional Info

release notes: [const Type Parameters](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#const-type-parameters)
Initial issue: https://github.com/eslint-functional/eslint-plugin-functional/issues/827

### PR DESCRIPTION
**Pull request by RebeccaStevens**
## PR Checklist

- [x] Addresses an existing open issue: fixes #518
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Initial attempt at resolving the issue.
I've got no idea if this is the correct approach or not.
There might be some cases which this PR doesn't currently handle.

### CODE REVIEW

File: typings/typescript.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import "typescript";
+
+declare module "typescript" {
+	// internal TS APIs
+
+	interface Symbol {
+		readonly links?: {
+			readonly checkFlags: ModifierFlags;
```
Comment (JoshuaKGoldberg) [inline typings/typescript.d.ts:8]: Yeah I can't find any other way to do this but don't _love_ relying on TS internals 😬 ... hmm. Do you think you could file an issue on TypeScript @RebeccaStevens asking to make this or an equivalent public?

**Comment by RebeccaStevens** [comment]
@JoshuaKGoldberg I just cleaned up this PR a bunch.

File: typings/typescript.d.ts (hunk 1 above)
Comment (RebeccaStevens) [reply typings/typescript.d.ts]: https://github.com/microsoft/TypeScript/issues/58656

**Review by JoshuaKGoldberg — APPROVED** [review]
Sorry for the delay, this looks good to me. Thanks!

**Comment by JoshuaKGoldberg** [comment]
Builds are very broken on `main` right now. I filed #530 to use the latest pnpm. Anyway, this has been ready for a while, and I just lost track of it 🤦

**Pull request merged**